projects/pt1/python/torch_mlir/jit_ir_importer/build_tools/abstract_interp_lib_gen.py

```diff
@@ -1831,6 +1851,20 @@ def aten〇reflection_pad1d〡dtype(self_rank_dtype: Tuple[int, int], padding: L
     assert len(padding) == 2, 'padding size expected to be 2'
     return self_dtype
 
+@check_dtype_function([ErrorInvocation(TensorOfShape(2, 3, 4), padding=1),
+                       ErrorInvocation(TensorOfShape(2, 3, 4), padding=[]),
+                       ErrorInvocation(TensorOfShape(2, 3, 4), padding=[2]),
+                       ErrorInvocation(TensorOfShape(2, 3, 4), padding=[2,1]),
+                       ErrorInvocation(TensorOfShape(2, 3, 4), padding=[3,2,1]),
+                       Invocation(TensorOfShape(5, 5, 4), padding=[1,2,3,4]),
+                       ErrorInvocation(TensorOfShape(2, 3, 4), padding=[5,4,3,2,1])])
+def aten〇reflection_pad2d〡dtype(self_rank_dtype: Tuple[int, int], padding: List[int]) -> int:
+    self_rank, self_dtype = self_rank_dtype
+
+    assert len(padding) == 4, 'padding size expected to be 4'
```

Comment:
Is this the right place for this assert? This seems closer to an op verification or a check during the lowering; the dtype should be independent of this.